optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/shift/ShiftRestService.java

```diff
@@ -58,18 +56,15 @@
 
     @PUT
     @Path("/template/update")
-    // TODO: Rename this updateTemplate
-    void createTemplate(@PathParam("tenantId") Integer tenantId, Collection<ShiftInfo> shifts);
+    void updateTemplate(@PathParam("tenantId") Integer tenantId, Collection<ShiftTemplate> shifts);
 
     @GET
     @Path("/template")
-    ShiftTemplate getTemplate(@PathParam("tenantId") Integer tenantId);
+    Collection<ShiftTemplate> getTemplate(@PathParam("tenantId") Integer tenantId);
```

Comment:
This should be a List<ShiftTemplate> with some order defined by the javadoc contract. I would expect sorted first by spot and then by startingDateTime?

Reply:
But what would we achieve by having them in a list? There is no performance gain or good use case from having them accessible by index. (The result from the query is a list, sorted by start time, then spot name, then rotation name, but that besides the point)

Reply:
Good point, but there is a general tendency in the java API's to expose List and Set over Collection. But they never expose ArrayList, HashSet, etc as that is too detailed. For some users, this makes it easier to get the first element of the Collection (list.get(0) vs collection.iterator().next()), last element of the Collection (list.get(size() - 1) vs ?), resort them etc.